bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java

```diff
@@ -197,40 +197,29 @@ public void initFromThingType(ThingTypeUID thingTypeUID) {
         isBlu = isBluSeries(thingTypeUID); // e.g. SBBT for BLU Button
         isGen2 = isGeneration2(thingTypeUID);
 
-        String type = getString(device.type);
-        isDimmer = type.equalsIgnoreCase(SHELLYDT_DIMMER) || type.equalsIgnoreCase(SHELLYDT_DIMMER2)
-                || type.equalsIgnoreCase(SHELLYDT_PLUSDIMMERUS) || THING_TYPE_SHELLYPLUSDIMMERUS.equals(thingTypeUID)
-                || THING_TYPE_SHELLYPLUSDIMMER10V.equals(thingTypeUID)
-                || THING_TYPE_SHELLYPLUSDIMMER.equals(thingTypeUID);
+        isDimmer = GROUP_DIMMER_THING_TYPES.contains(thingTypeUID);
         isBulb = THING_TYPE_SHELLYBULB.equals(thingTypeUID);
-        isDuo = THING_TYPE_SHELLYDUO.equals(thingTypeUID) || THING_TYPE_SHELLYVINTAGE.equals(thingTypeUID)
-                || THING_TYPE_SHELLYDUORGBW.equals(thingTypeUID);
-        isRGBW2 = THING_TYPE_SHELLYRGBW2_COLOR.equals(thingTypeUID) || THING_TYPE_SHELLYRGBW2_WHITE.equals(thingTypeUID)
-                || THING_TYPE_SHELLYPLUSRGBWPM.equals(thingTypeUID);
+        isDuo = GROUP_DUO_THING_TYPES.contains(thingTypeUID);
+        isRGBW2 = GROUP_RGBW2_THING_TYPES.contains(thingTypeUID);
         isLight = isBulb || isDuo || isRGBW2;
```

Comment:
This logic is now duplicated, so for consideration:
```suggestion
        isLight = GROUP_LIGHT_THING_TYPES.contains(thingTypeUID);
```

Reply:
changed